Shorter Heat Equation is Hello World of CSE Blog article

markcmiller86 opened the pull request.

gonsie commented on Articles/Blog/HelloHeatEquation.md:
```diff
@@ -1,143 +1,99 @@
+
+## A Hello CSE World Repository
+We are announcing the beginnings of an effort to populate and maintain a
+repository of implementations of the heat equation demonstrating various
```
include a link to the repository

markcmiller86 replied:
Done. Also see [this PR](https://github.com/betterscientificsoftware/hello-heat-equation/pull/3)

gonsie commented on Articles/Blog/HelloHeatEquation.md:
```diff
@@ -1,143 +1,99 @@
-Using the Heat Equation to demonstrate the power of a programming model or mathematical library is not new. In fact, it is common tutorial exercise for new scientific programmers. Since there are too many interesting examples to include in this article, we've started [a repository showcase](https://github.com/betterscientificsoftware/hello-heat-equation) to collect the many examples.
+## Why this form of the Heat Equation?
+While the one dimensional heat equation might not be sufficient to
+model solar heating of the Earth, even in this relatively simple form,
```
run-on sentence?

markcmiller86 replied:
fixed

gonsie commented on Articles/Blog/HelloHeatEquation.md:
```diff
@@ -1,143 +1,99 @@
+the scientific software community. Collecting a number of working examples
+together with unique implementation characteristics will help to create 
+useful introductory CSE training material. We welcome collaboration.
+To get involved, please contact one of the authors.
```
Details on getting involved can be found in the repo's contributing guide, please include a link. Remove 'contact one of the authors'.

gonsie commented on Articles/Blog/HelloHeatEquation.md:
```diff
@@ -1,143 +1,99 @@
+demonstrating the use of ECP-CI resources.
+
+Each implementation we gather into this repository will have to conform
+to a *minimal* set of requirements including providing a fully self-contained
```
We are not asking the examples to be runnable, only that they use common naming scheme for variables.

markcmiller86 replied:
As per side-bar conversation at DOE Data Day, we agreed to keep snipits for now in the repo created for that purpose. Eventually, those will be replaced with a repo hosting fully working examples of same or similar lineage.

gonsie commented on Articles/Blog/HelloHeatEquation.md:
```diff
@@ -1,143 +1,99 @@
+Our goal in collecting such examples isn't so much to demonstrate the use of
+different programming languages as it is to demonstrate different approaches
+to computation and execution paradigms as well as provide instructive insights
+by comparing of time, space and scaling performance of various approaches.
```
We do not compare time, space, and scaling performance.

markcmiller86 replied:
As per conversation at DOE Data Day, the ultimate intention is to do so and its ok for this Blog article to announce that.

gonsie commented on Articles/Blog/HelloHeatEquation.md:
```diff
@@ -1,143 +1,99 @@
+* Apart from initial/boundary conditions, there are no other heat sources
 * Only Cartesian coordinates in 1 dimension are used
 
-Since the heat equation takes the form of a PDE over a continuous space, most programs create a discretization of the formula over a finite set of sampling points.
```
please move the deleted text to an explanation page in the [hello-heat-equation repository](https://github.com/betterscientificsoftware/hello-heat-equation)

markcmiller86 replied:
Done. See [this PR](https://github.com/betterscientificsoftware/hello-heat-equation/pull/3)

gonsie commented on Articles/Blog/HelloHeatEquation.md:
```diff
@@ -1,143 +1,99 @@
+in a material. While there are many adaptations for different scenarios, beginner CSE
+programmers often start with the one-dimensional, homogeneous Heat Equation as an
+introductory problem.
 
```
please add an introduction to the equation. Something like "The general heat equation is:"

markcmiller86 commented:
@gonsie as per conversation, I moved text I had previously wholly deleted to the companion code snipits repository's `README.md` file. That change is in [this PR](https://github.com/betterscientificsoftware/hello-heat-equation/pull/3). I added a line here with a link to that repo and a short explanation as to that repo's purpose. In addition, I addressed the other requests you made here.

markcmiller86 commented:
> Is there an official style guide for BSSW?

@gonsie this is a great question. Not to my knowledge. @rinkug?

> I would prefer to use the Oxford comma.

I had to go look up what *Oxford comma* meant. It is the comma before the *and* in a list of items as in the comma after *strawberries*.

> I like bananas, strawberries, and peaches.

FWIW, that is not my particular style preference but seeing as how you and @heatherms27 are lead authors, I will do my best to go back and correct where appropriate.

markcmiller86 commented:
> FWIW, that is not my particular style preference but seeing as how you and @Heatherms27 are lead authors, I will do my best to go back and correct where appropriate.

@gonsie that is done now. So, just awaiting your final review.

rinkug commented:
This  is PR for issue #397

rinkug commented:
We need a reviewer for this.

rinkug commented:
Please refer to the next PR #425 for discussion continuation
